src/core/thread/mle.hpp

```diff
@@ -1582,6 +1582,13 @@ class Mle : public InstanceLocator
         kParentRequestTypeRoutersAndReeds, ///< Parent Request to all routers and REEDs.
     };
 
+    enum ChildUpdateRequestPendingStatus
+    {
+        kChildUpdateRequestPendingNo,
+        kChildUpdateRequestPendingInProgress,
+        kChildUpdateRequestPendingInit,
```

Comment:
nit: what about
```c++
enum ChildUpdateRequestState
{
    kChildUpdateRequestNone,
    kChildUpdateRequestPending,
    kChildUpdateRequestActive,
};
```

I also tend to prefer to list the states in the order that they typically occur.

Reply:
updated.